test/test.pl

```diff
@@ -1964,6 +1980,15 @@ sub test_view
          ['-R', $fogn, '-r', 'grp2'], 0],
         ['rg_both2', { read_groups => { grp1 => 1, grp2 => 1, grp3 => 1 }},
          ['-r', 'grp2', '-R', $fogn], 0],
+        # Barcodes
+        ['bc_TGCA', { barcode_tag => 'BC', barcodes => { TGCA => 1 }},
+         ['-d', 'BC:TGCA'], 0],
+        ['bc_fobc', { barcode_tag => 'BC', barcodes => { ACGT => 1, AATTCCGG => 1 }},
+         ['-D', "BC:${fobc}"], 0],
+        ['bc_both', { barcode_tag => 'BC', barcodes => { ACGT => 1, TGCA => 1, AATTCCGG => 1 }},
+         ['-D', "BC:${fobc}", '-d', 'BC:TGCA'], 0],
```

Comment:
I was under the impression the two modes were mutually exclusive, at least in practice. If not, there should be different holders for the two barcode tags in `settings`, as currently the barcode tag from the single mode would overwrite the barcode tag from the file mode, causing a memory leak and unexpected results.

Reply:
@valeriuo they are not mutually exclusive. I now check if the same barcode tag is provided.

It also allows to specify the options multiple times.
```
/samtools view -d BC:ACGT -d BC:AATTCCGG test/dat/view.001.sam
```